### ISSUE TITLE
driver-kubelet-plugin : error running nvidia-smi: signal: segmentation fault (core dumped)

### ISSUE THREAD

**Issue by CoderTH** [NVIDIA/k8s-dra-driver#7]
When I apply -f demo1.yaml and find that the pod is always in the create state, kubelet-plugin gets the following error. It seems to be an nvidia-smi error?
![image](https://github.com/NVIDIA/k8s-dra-driver/assets/51119718/00605690-8c15-4011-baff-62ae1d4bb8d4)

I tried running nvidia-s'mi manually into the container and it was fine
![image](https://github.com/NVIDIA/k8s-dra-driver/assets/51119718/960627ac-a8d2-4362-a18c-d0118dedeb08)

I guessed that this error was related to the following code, so I simulated running the relevant commands, and it was normal.

```
func (t *TimeSlicingManager) SetTimeSlice(devices *PreparedDevices, config *nascrd.TimeSlicingConfig) error {
	if devices.Mig != nil {
		return fmt.Errorf("setting a TimeSlice duration on MIG devices is unsupported")
	}

	timeSlice := nascrd.DefaultTimeSlice
	if config != nil && config.TimeSlice != nil {
		timeSlice = *config.TimeSlice
	}

	err := t.nvdevlib.setComputeMode(devices.UUIDs(), "DEFAULT")
	if err != nil {
		return fmt.Errorf("error setting compute mode: %w", err)
	}

	err = t.nvdevlib.setTimeSlice(devices.UUIDs(), timeSlice.Int())
	if err != nil {
		return fmt.Errorf("error setting time slice: %w", err)
	}

	return nil
}
```

![image](https://github.com/NVIDIA/k8s-dra-driver/assets/51119718/5cf8dbf7-d637-4173-bf25-0a2abc0678c7)

The os versions I use are centos7.9 and tesla p4

**Comment by elezar** [comment]
@CoderTH with the changes that you are testing the equivalent command is no longer `chroot /driver-root nvidia-smi`.

Instead, we update the `PATH` to include the location of `nvidia-smi` in the driver root and set `LD_LIBRARY_PATH` to include the path to `libnvidia-ml.so.1` which is required by `nvidia-smi`. It could be that this configuration is causing a different library to be loaded when running `nvidia-smi` to set the compute mode.

Could you check the behaviour when:
1. Exec-ing into the container
2. Setting `PATH` to include the path (in `/driver-root`) to `nvidia-smi` at the start
3. Setting `LD_LIBRARY_PATH` to include the path (in `/driver-root`) at the start

And then running the relevant `nvidia-smi` commands.

**Comment by CoderTH** [comment]
> @CoderTH with the changes that you are testing the equivalent command is no longer `chroot /driver-root nvidia-smi`.
> 
> Instead, we update the `PATH` to include the location of `nvidia-smi` in the driver root and set `LD_LIBRARY_PATH` to include the path to `libnvidia-ml.so.1` which is required by `nvidia-smi`. It could be that this configuration is causing a different library to be loaded when running `nvidia-smi` to set the compute mode.
> 
> Could you check the behaviour when:
> 
> 1. Exec-ing into the container
> 2. Setting `PATH` to include the path (in `/driver-root`) to `nvidia-smi` at the start
> 3. Setting `LD_LIBRARY_PATH` to include the path (in `/driver-root`) at the start
> 
> And then running the relevant `nvidia-smi` commands.
> 
> My assumption is that some library in `LD_LIBRARY_PATH` is conflicting with a library in the container.

Yes, with the method you said, I successfully reproduced the error in logs

![image](https://github.com/NVIDIA/k8s-dra-driver/assets/51119718/5395af5e-ec50-4b1f-b0f7-99bbe734531f)

**Comment by CoderTH** [comment]
> LD_LIBRARY_PATH

When I installed strace and set /driver-root in PATH and LD_LIBRARY_PATH, strace didn't work with the following error

<img width="1008" alt="image" src="https://github.com/NVIDIA/k8s-dra-driver/assets/51119718/c3a1c86d-947b-477a-99fc-806339f9be8c">
When I removed /driver-smi, it worked, but nvidia-smi couldn't find the libnvidia-mi.so library

**Comment by elezar** [comment]
This means that `libc.so.6` is being resolved from the host in the the container which may be causing this problem. Since this plugin is running in an ubuntu container, it would make sense to prepend the location of `libc.so.6` to `LD_LIBRARY_PATH` to ensure that this is resolved in the container instead.

```
dirname $(find / | grep libc.so)
/lib/x86_64-linux-gnu
```

There may be other libraries that are also being aliased. What are the contents of `/driver-root/lib64` in your case?

**Comment by elezar** [comment]
See https://gitlab.com/nvidia/cloud-native/k8s-dra-driver/-/merge_requests/26 with a proposed workaround / solution for this.

**Comment by elezar** [comment]
@CoderTH just for completeness, could you run `ldd nvidia-smi` when having an incorrect path set up?

**Comment by elezar** [comment]
@CoderTH this should have been addressed in https://github.com/NVIDIA/k8s-dra-driver/commit/c53420ebee2eee68d56699f0e59391b90d026b3c could you confirm that this is the case?

**Comment by CoderTH** [comment]
> @CoderTH this should have been addressed in [c53420e](https://github.com/NVIDIA/k8s-dra-driver/commit/c53420ebee2eee68d56699f0e59391b90d026b3c) could you confirm that this is the case?

I am very sorry for the late reply. Some other things have occupied my time recently. I have re-tested with your latest commit today and it has failed to report segmentation fault, but new errors have been found. Does this error refer to the fact that my tesla p4 does not support timeslice model GPU sharing?


![image](https://github.com/NVIDIA/k8s-dra-driver/assets/51119718/c3724682-b9d8-4a6d-97bd-938e4dade2bd)
